### DISCUSSION TITLE
Path options for reading and writing PnPSite Templates

### DISCUSSION THREAD

**Discussion by tandy-wine** [pnp/powershell#1001] [Q&A]
I have a question about the usage of the PnP commands and the path of the template.  The following command works to create a template of the site I'm connected to:
     Get-PnPSiteTemplate  -out “C:\temp\TestTemplate.xml”

Then I can apply the template by connecting to a new site and running this: 
     Invoke-PnPSiteTemplate -Path “C:\temp\TestTemplate.xml”

But can I specify a template that is saved into OneDrive or into a document library in a Sharepoint site?  Or can I only write/read the template file to a mapped drive (such as the c: drive)?  I'm trying to figure out where to store templates so that I and others can apply the same template to standardize after creating a Sharepoint site.

**Comment by BaronSparky** [comment]
Hi. You can invoke SiteTemplates directly from a URL. i.e. store the .XML or .PNP file in a document library, and invoke from there to your current context.

You cannot do the same for TenantTemplates however. They have to be enacted from Windows :(

**Reply by tandy-wine** [reply]
Thank you for the reply.  Can you help me with the syntax?  Instead of:
      Get-PnPSiteTemplate -out “C:\temp\TestTemplate.xml”
Are you suggesting that the following should work?
    Get-PnPSiteTemplate -out “https://mytenant.sharepoint.com/sites/MySite/TemplateLibrary/TestTemplate.xml”

This doesn't work for me.  I get the error:  "Get-PnPSiteTemplate : The given path's format is not supported"

**Comment by BaronSparky** [comment]
Hi. I was referring to the second part of your question :)

You can store SiteTemplates in a document library and "Invoke-PnPSiteTemplate" directly from that library. 
[https://pnp.github.io/powershell/cmdlets/Invoke-PnPSiteTemplate.html](url) - "Example 6"
You would have to copy the template into that library first though. I believe the "Get-" requires a windows path.

**Reply by tandy-wine** [reply]
Thank you for your reply.  I understand and have tried this.  But unfortunately I'm been creating templates as XML files and have not used the PNP format.  So it appears the Invoke command can only pull the template file from a sharepoint document library if the format is PNP and not XML.
